src/rdkafka_roundrobin_assignor.c

```diff
@@ -75,21 +75,37 @@ rd_kafka_roundrobin_assignor_assign_cb (rd_kafka_t *rk,
                 rd_kafka_assignor_topic_t *eligible_topic = eligible_topics[ti];
 		int partition;
 
+		/* Sort eligible members by name */
+		rd_list_sort(&eligible_topic->members,
+			     rd_kafka_group_member_cmp);
+
 		/* For each topic+partition, assign one member (in a cyclic
 		 * iteration) per partition until the partitions are exhausted*/
 		for (partition = 0 ;
 		     partition < eligible_topic->metadata->partition_cnt ;
 		     partition++) {
-			rd_kafka_group_member_t *rkgm;
+			rd_kafka_group_member_t *next_rkgm = &members[next];
+
+			/* Find if next member is eligible */
```

Comment:
Is all this logic really needed? Can't we just start over at next=0 if find_subscription() exhausts all members?
We know that there is at least one member in the list that is subscribing to this topic.